docs/src/main/paradox/user/lang/kotlin/eventsourced.md

```diff
@@ -0,0 +1,103 @@
+
+Event handlers are differentiated by the type of event they handle. By default, the type of event an event handler handles will be determined by looking for a single non context parameter that the event handler takes. If for any reason this needs to be overridden, or if the event handler method doesn't take any non context parameter (because the event type may be all that needs to be known to handle the event), the type of event the handler handles can be specified using the `eventClass` parameter on the `@EventHandler` annotation.
+
+Event handlers may be declared for a superclass or interface of the types they handle, for example an event handler that declares an `Object` parameter will handle all events. In the case where multiple event handlers match, Cloudstate will match the most specific event handler, which is decided by walking up the superclass tree, and matching all interfaces along the way.
+
+Here's an example event handler for the `ItemAdded` event. A utility method, `convert` is also defined to assist it.
+
+@@snip [ShoppingCartEntity.kt](/docs/src/test/kotlin/docs/user/eventsourced/ShoppingCartEntity.kt) { #item-added }
+
+## Producing and handling snapshots
+
+Snapshots are an important optimisation for event sourced entities that may contain many events, to ensure that they can be loaded quickly even when they have very long journals. To produce a snapshot, a method annotated with `@Snapshot` (io.cloudstate.kotlinsupport.api.eventsourced.Snapshot) must be declared. It takes a context class of type `SnapshotContext` (io.cloudstate.javasupport.eventsourced.SnapshotContext), and must return a snapshot of the current state in serializable form. 
+
+@@snip [ShoppingCartEntity.kt](/docs/src/test/kotlin/docs/user/eventsourced/ShoppingCartEntity.kt) { #snapshot }
+
+When the entity is loaded again, the snapshot will first be loaded before any other events are received, and passed to a snapshot handler. Snapshot handlers are declare by annotating a method with `@SnapshotHandler` (io.cloudstate.kotlinsupport.api.eventsourced.SnapshotHandler), and it can take a context class of type `SnapshotContext` (io.cloudstate.javasupport.eventsourced.SnapshotContext).
+
+Multiple snapshot handlers may be defined to handle multiple different types of snapshots, the type matching is done in the same way as for events.
+
+@@snip [ShoppingCartEntity.kt](/docs/src/test/kotlin/docs/user/eventsourced/ShoppingCartEntity.kt) { #handle-snapshot }
+
+## Registering the entity
+
+Once you've created your entity, you can register it with the `cloudState` (io.cloudstate.kotlinsupport.cloudState) server, by invoking the `registerEventSourcedEntity` function. In addition to passing your entity class and service descriptor, you also need to pass any descriptors that you use for persisting events, for example, the `domain.proto` descriptor.
```

Comment:
in the code, this is all lowercase:

https://github.com/cloudstateio/kotlin-support/blob/0aae079f1904a3a64e34aa9d51accaef3ea86387/cloudstate-kotlin-support/src/main/kotlin/io/cloudstate/kotlinsupport/cloudstate.kt#L6

Reply:
Yes this is a high order function. Not a Class.